ENIGMAsystem/SHELL/Widget_Systems/Win32/dialogs.cpp

```diff
@@ -55,9 +55,9 @@ static string gs_but1, gs_but2, gs_but3;
 void show_error(string errortext, const bool fatal)
 {
   #ifdef DEBUG_MODE
-    errortext += enigma::debug_scope::GetErrors();
+  errortext = enigma::debug_scope::GetErrors() + "\r\n\r\n" + errortext;
```

Comment:
This is not related to `get_directory` and should be split or explained.

Reply:
I did this because it is a change made in my other pr and i didn't want any conflcts as I intend to get the other one merged first so no changes get undone.

Reply:
That being said, I'll let you know when Josh merges my other pr before this one is ready to be.

Reply:
Ok, that's fine, but your concerns are unwarranted. Git simply doesn't work like that. The commit just records "inserted these lines at this position" and would raise a merge conflict whenever two commits tried to insert to the same place. Just in general, unless you had a reason to include this here (maybe because the other stuff was dependent), I don't think it's a good idea to do this and it may just end up causing you more grief than it's worth. But, it's fine, I am not going to notice if the other one is merged.

Reply:
Thank you. And sorry for the flood of retarded messages on hangouts. It's not even worth reading at this point.

Reply:
Sure, just stay on topic is all I ask, I'm very stressed right now. Thanks.

Reply:
Leaving this one unresolved until you talk to Josh about that other pull request.

Reply:
Sounds good.

Reply:
@RobertBColton alright, Josh mudged my other pull request, so i guess that means you can mudge this one too :) Hoorah for mudging!!!

Reply:
Two things:

1) I disagree with the way GM8 showed error messages. Usually the error is shown first and the stack is shown second in an error message dialog because stacks can be very huge. It causes the error messages to get buried, and quite frankly, I don't really even think we need to add the extra text for run mode either.
2) This follows from above, the debug scope `GetErrors` actually has a bug. It starts with an extra newline when it doesn't need to. Also it prints the stack from top to bottom, not top to bottom. So because of (1) the stack now looks backwards. This is an error message I show in a script, but the stack looks like it occurred in the object create event.
![image](https://user-images.githubusercontent.com/3212801/55045859-d5063580-5015-11e9-8638-7d624d8a5aeb.png)

In my opinion, this change should just be undone and `GetErrors` should be changed to put the new line on the end of the line rather than the first character.

Reply:
Here's what I think it should look like:
![ENIGMA Suggested Stack Trace Dialog](https://user-images.githubusercontent.com/3212801/55046695-d5540000-5018-11e9-9dfd-73c4fb4b61b8.png)

Reply:
@RobertBColton Sure, will do. Would you like me to reflect these changes in our other widgets systems or should that be a separate pr?

Reply:
Up to you, here's a couple of things to mention. I am not entirely sure my way is perfect, just objectively good.
* I copied the above look from Python actually.
* I guess you could say showing the error stack from start to end first (like <GM8) is maybe easier for novices to understand? In the same way update events are named "step", the novice user can think of it like stepping through the program to see where there game had an error, then finally they see the error. That seems straight forward to a non-computer scientist to me (I mean we read code like a recipe top to bottom).
* However, printing a stack bottom to top like GM8 did actually requires `GetErrors` to use a second temporary stack to unwind and then build the errors string. You know, `std::stack` doesn't even allow sequential access.
* Finally, printing a stack like GM8 did is super inconsistent with a lot of stuff and I'm afraid it'll be confusing. Python prints it most recent frame last like GM8. But you know GDB's backtrace command prints it most recent frame first. Java also prints it most recent frame first. To be honest though, a lot of IDEs let you just change the sorting order of a stack trace in the UI, so it's not too big of a deal.

Anyway, to answer your question, that's basically up to you. If you agree with me here, then yes. I think the argument is there that what I've suggested is objectively better than GM8.

Reply:
I agree, so yeah I'll update the other systems.

Reply:
Here's debug scope's location if you're looking for it:
https://github.com/enigma-dev/enigma-dev/blob/b074d6bf06fd7aaea9d4a137a5a0c4f7bf2226f9/ENIGMAsystem/SHELL/Universal_System/debugscope.cpp#L41

Reply:
Thanks!